### ISSUE TITLE
Can't able to train the agent using stable baselines ?

### ISSUE THREAD

**Issue by kunalrawat2000** [maximecb/gym-minigrid#144]
Hello maximecb,
Thanks for this amazing repo, I'm making a custom environment using the mini-grid where I have added obstacles so that whenever an agent crosses them the reward they will get is -1 else zero and at the end when it reaches the goal the reward will be 1 and while training the agent with stable baselines(ppo and a2c) the maximum reward I have achieved till now is -50 means it is overlapping with the  obstacles instead of avoiding them and not learning properly 
obstacle class:
```
class nfa(WorldObj):
    def __init__(self, color='purple'):
        super().__init__('ball', color)
    
    def can_overlap(self):
        """Can the agent overlap with this?"""
        return True

    def render(self, img):
        fill_coords(img, point_in_rect(0, 1, 0, 1), COLORS[self.color])
```
Custom Environment:
![image](https://user-images.githubusercontent.com/54204821/125774654-1ba04f98-ceac-4ec3-90da-026725682845.png)

PPO Parameters:

```
l_rate = 0.0007   #0.0003
batch = 64         #64
df = 0.99         #0.99
n_steps = 2048    #2048


policy_kwargs = dict(activation_fn=th.nn.ReLU,
                     net_arch=[ 32,64,64,128])


model = PPO( 'MlpPolicy' , 
            env, 
            verbose =2   
            ,learning_rate = l_rate 
            ,batch_size = batch 
            ,gamma=df 
            ,n_steps = n_steps
            ,clip_range = 0.2
            ,policy_kwargs=policy_kwargs
            )   
```

**So can you suggest me what should I do? which parameters to tune or should I use some other framework other than stable baselines3?**

**Comment by maximecb** [comment]
I would suggest starting with a much smaller environment eg 6x6 and seeing if you can get that to converge. Optimize your parameters, then move to larger environments.

**Comment by kunalrawat2000** [comment]
If the obstacles are static then it is working fine but I'm choosing the obstacles randomly, so I tried with a smaller grid but the problem is still there.

**Comment by maximecb** [comment]
Two things to try: give your model some recurrent memory, and try giving a positive reward for succeeding but no negative reward on failure. In my experience negative rewards work poorly. Besides that, keep experimenting with small envs like 6x6 or 8x8 until you figure out a recipe that works before moving to larger ones.
